sql/expression/matchagainst.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
```

Comment:
I think given the structure of what we're trying to accomplish, this can't be (only) an expression node. The analyzer needs to translate this expression into a (set of?) plan node(s) which does the index lookups and only returns the relevant rows, right?

There can definitely be an expression component which is filtering the returned (un?)projected rows based on the initial cast of the index lookups though...